.github/workflows/rust_tests.yml

```diff
@@ -0,0 +1,54 @@
+name: Rust Tests
```

Comment:
Github Actions for CI Testing!

We are running:
- `cargo test` to run all the Rust Tests that we will write
- `cargo clippy` which is the Rust linter
- `cargo deny check bans` which will look for Crate (Rust packages) that I've explicitly banned. This is really just a ban on `openssl` right now, since that a `C` library and makes deploying more annoying than it needs to be. So we try to always use `rustls` which is a full Rust implementation. And this check helps make sure openssl doesn't sneak in!


`cargo test` and `cargo clippy` are the things you may want to run locally! The deny don't worry about too much but if it triggers in a PR where you add a crate I'll dig into it with ya!